tests/testthat/test-infection-integration.R

```diff
@@ -253,10 +252,33 @@ test_that('calculate_clinical_infections correctly samples clinically infected',
 })
 
 test_that('calculate_treated correctly samples treated and updates the drug state', {
+  
```

Comment:
It is best to keep existing tests relatively unchanged so that it's clear to the reviewers that the new changes won't affect existing behaviour.

Could you revert this test, copy it into a new test, rename it and then test resistance behaviour in that new test.

Reply:
I have reverted the test to it's original version and added a separate version of the same test for the antimalarial resistance behaviour.